src/main/java/org/opensearch/performanceanalyzer/rca/store/rca/admissioncontrol/heap/MediumHeapRca.java

```diff
@@ -0,0 +1,120 @@
+    private double previousHeapPercent = 0;
+
+    private InstanceDetails instanceDetails;
+    private RangeConfiguration requestSizeHeapRange;
+
+    @Override
+    public void init(InstanceDetails instanceDetails, RangeConfiguration rangeConfiguration) {
+        this.instanceDetails = instanceDetails;
+        this.requestSizeHeapRange = rangeConfiguration;
+    }
+
+    @Override
+    public ResourceFlowUnit<HotNodeSummary> operate(HeapMetric heapMetric) {
+
+        long currentTimeMillis = System.currentTimeMillis();
+        double currentHeapPercent = heapMetric.getHeapPercent();
+
+        HotNodeSummary nodeSummary =
+                new HotNodeSummary(
+                        instanceDetails.getInstanceId(), instanceDetails.getInstanceIp());
+
+        // If we observe heap percent range change then we tune request-size controller threshold
+        // by marking resource as unhealthy and setting desired value as configured
+        if (requestSizeHeapRange.hasRangeChanged(previousHeapPercent, currentHeapPercent)) {
+
+            double currentThreshold = getThreshold(requestSizeHeapRange, currentHeapPercent);
+            if (currentThreshold == 0) {
+                // AdmissionControl rejects all requests if threshold is set to 0, thus ignoring
+                return new ResourceFlowUnit<>(
+                        currentTimeMillis,
+                        new ResourceContext(HEALTHY),
+                        nodeSummary,
+                        !instanceDetails.getIsMaster());
+            }
+
+            LOG.debug(
+                    "[AdmissionControl] PreviousHeapPercent={} CurrentHeapPercent={} CurrentDesiredThreshold={}",
+                    previousHeapPercent,
+                    currentHeapPercent,
+                    currentThreshold);
+
+            double previousThreshold = getThreshold(requestSizeHeapRange, previousHeapPercent);
+            previousHeapPercent = currentHeapPercent;
+
+            HotResourceSummary resourceSummary =
+                    new HotResourceSummary(HEAP_MAX_SIZE, currentThreshold, previousThreshold, 0);
+            nodeSummary.appendNestedSummary(resourceSummary);
+
+            RCA_VERTICES_METRICS_AGGREGATOR.updateStat(
+                    ADMISSION_CONTROL_RCA_TRIGGERED, instanceDetails.getInstanceId().toString(), 1);
+
+            return new ResourceFlowUnit<>(
+                    currentTimeMillis,
+                    new ResourceContext(UNHEALTHY),
+                    nodeSummary,
+                    !instanceDetails.getIsMaster());
+        }
+
+        return new ResourceFlowUnit<>(
+                currentTimeMillis,
+                new ResourceContext(HEALTHY),
+                nodeSummary,
+                !instanceDetails.getIsMaster());
+    }
+
+    private double getThreshold(RangeConfiguration heapRange, double heapPercent) {
+        Range range = heapRange.getRange(heapPercent);
+        return Objects.isNull(range) ? 0 : range.getThreshold();
```

Comment:
I think there are more spotless errors, can we fix them?

Reply:
Successfully completes `./gradlew build`